### PR TITLE
Remapfile

### PR DESCRIPTION
**Pull request by sfranky**
remap mappings inputtable in qtopconf.yaml

### CODE REVIEW

**Comment by sfranky** [comment]
unmapped mapped mappings

File: colormap.py [hunk 1]
```diff
@@ -123,7 +123,6 @@
     'cmsau': 'Green',
     'twcms': 'Green',
     'cmsmu': 'Green',
-    'cms': 'Green',
```
Comment (fgeorgatos) [inline colormap.py]: is this a duplicate of something or what?

File: pbs.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def make_pbsnodes(orig_file, out_file, write_method):
             pbs_values['core_job_map'] = []
             jobs = block['jobs'].split(',')
             for job, core in get_jobs_cores(jobs):
-                _d = {}
+                _d = dict()
```
Comment (fgeorgatos) [inline pbs.py]: explain the difference?

File: qtop.py [hunk 3]
```diff
@@ -11,9 +11,9 @@
 from optparse import OptionParser
 import datetime
 from collections import Counter, OrderedDict
-import os
-import re
-import yaml
+# import os
+# import re
+# import yaml
```
Comment (fgeorgatos) [inline qtop.py]: why not simply drop them and rely on git commits about it? r u concerned it won't be easy to revert?

File: pbs.py (hunk 2 above)
Comment (sfranky) [inline pbs.py]: no difference (not that i know of), is preferred over {} by pycharm in its suggestions. (explicit better than implicit??)

File: qtop.py [hunk 4]
```diff
@@ -47,17 +47,16 @@
...
     """
     try:
-        color_code = code_of_color[color_of_account[pattern]]
+        color_code = code_of_color[color] if color else code_of_color[color_of_account[pattern]]
     except KeyError:
         return text
     else:
-        return "\033[" + color_code + "m" + text + "\033[1;m" if ((not options.NOCOLOR) and pattern != 'account_not_coloured') \
-            else text
+        return "\033[" + color_code + "m" + text + "\033[1;m" if ((not options.NOCOLOR) and pattern != 'account_not_coloured' and text != ' ') else text
```
Comment (fgeorgatos) [inline qtop.py]: long line, try to find a way to make it sorter

File: colormap.py (hunk 1 above)
Comment (sfranky) [inline colormap.py]: i think this is from the original colormap file, but i could be wrong

File: qtop.py [hunk 5]
```diff
@@ -149,7 +147,22 @@ def calculate_stuff(pbs_nodes):
     # is this even needed anymore?!
     for i in range(1, node_dict['highest_wn'] + 1):
         if i not in node_dict['wn_dict']:
-            node_dict['wn_dict'][i] = {'state': '?', 'np': 0}  # was: node_dict['wn_dict'][i] = '?'
+            node_dict['wn_dict'][i] = {'state': '?', 'np': 0, 'domainname': 'N/A', 'host': 'N/A'}
+            # was: node_dict['wn_dict'][ i] = '?'
+
+    for _, state_corejob_dn in node_dict['wn_dict'].items():
+        _host = state_corejob_dn['domainname'].split('.', 1)[0]
+        changed = 0
```
Comment (fgeorgatos) [inline qtop.py]: true, false? that's boolean, isn't it?

File: qtop.py (hunk 3 above)
Comment (sfranky) [inline qtop.py]: not yet convinced i don't need them (they *should* be in use!). Also, don't know how to revert as quick as I can uncomment

File: qtop.py (hunk 5 above)
Comment (sfranky) [inline qtop.py]: to be fixed

File: qtop.py [hunk 6]
```diff
@@ -435,23 +449,38 @@ def print_wnid_lines(start, stop, highest_wn, hxxxx):
...
+        # node_str_width = len(max(hosts, key=len))
+
+        appends = {
+            '1': ['={__WNID__}'],
+            '2': ['={_Worker_}', '={__Node__}'],
+            '3': ['={_Worker_}', '={__Node__}', '={___ID___}'],
+            '4': ['={________}', '={_Worker_}', '={__Node__}', '={___ID___}'],
+            '5': ['={__FOR___}', '={__SALE__}', '={_Worker_}', '={__Node__}', '={___ID___}'],
+            '6': ['={_SPACE__}', '={__FOR___}', '={__SALE__}', '={_Worker_}', '={__Node__}', '={___ID___}'],
+            '7': ['={__PLACE_}', '={__YOUR__}', '={_ADVERT_}', '={__HERE__}', '={_Worker_}', '={__Node__}', '={___ID___}'],
```
Comment (fgeorgatos) [inline qtop.py]: ;-)

File: qtop.py [hunk 7]
```diff
@@ -435,23 +449,38 @@ def print_wnid_lines(start, stop, highest_wn, hxxxx):
...
+        }
+        for num in range(8, len(wn_vert_labels) + 1):
+            appends.setdefault(str(num), appends['7'] + num * ['={___ID___}'])
+        size = str(len(wn_vert_labels))  # key, nr of horizontal lines to be displayed
+        end_label = iter(appends[size])
+        for line_nr in wn_vert_labels:
```
Comment (fgeorgatos) [inline qtop.py]: too dense the above lines, some comments would help

File: qtop.py [hunk 8]
```diff
@@ -435,23 +449,38 @@ def print_wnid_lines(start, stop, highest_wn, hxxxx):
...
+            highlight = highlight_alternately(*config['alt_label_highlight_colours'])
```
Comment (fgeorgatos) [inline qtop.py]: alternatively?

File: qtop.py [hunk 9]
```diff
@@ -701,7 +725,7 @@ def calculate_split_screen_size():
 
     pbs_nodes = read_pbsnodes_yaml(PBSNODES_OUT_FN, options.write_method)
     total_running, total_queued, qstatq_lod = read_qstatq_yaml(QSTATQ_OUT_FN, options.write_method)
-    job_ids, user_names, job_states, _ = read_qstat_yaml(QSTAT_OUT_FN, options.write_method)  #_ == queue_names, not used for now
+    job_ids, user_names, job_states, _ = read_qstat_yaml(QSTAT_OUT_FN, options.write_method)  # _ == queue_names, not used for now
```
Comment (fgeorgatos) [inline qtop.py]: long line?

File: qtopconf.yaml [hunk 10]
```diff
@@ -4,15 +4,23 @@ separator: '|' # character used for vertical separator, for the -o switch
...
+ - '\w*atl\w+': Cyan_L
+wn_labels_max_len: 15
+alt_label_highlight_colours: [White, Blue_L]
+remapping:
+ - "([A-Za-z]+)([1-9]|[1-9][0-9]|[0-9][0-9][0-9])$": |
+     lambda m: str(m.group(1)+str(int(m.group(2))+250))
+ - 'torvalds(\d+)' : 'tor\1'
+ - 'gridlab(\d+)' : 'glab\1'
+ - 'gridmon(\d*)' : 'mon\1'
+ - '-' : ''
```
Comment (fgeorgatos) [inline qtopconf.yaml]: OK, it all looks good, just explain what this last line is

Comment (sfranky) [inline qtopconf.yaml]: my-wn-001 --> mywn001 
shortens it

File: qtop.py (hunk 6 above)
Comment (fgeorgatos) [inline qtop.py]: shouldn't this be configurable in a file, rather have hardwired strings as part of the code?

Comment (sfranky) [inline qtop.py]: definitely!

File: qtopconf.yaml (hunk 10 above)
Comment (fgeorgatos) [inline qtopconf.yaml]: ok, make this a comment, it would help the next guy reading it

**Pull request merged**